bluesky/utils.py

```
@@ -274,21 +275,25 @@ class Struct(metaclass=StructMeta):
     _fields = []
 
     def __init__(self, *args, **kwargs):
+        # Now bind default values of optional arguments.
+        # If it seems like there should be a cleaner way to do this, see
+        # http://bugs.python.org/msg221104
         bound = self.__signature__.bind(*args, **kwargs)
+        for name, param in self.__signature__.parameters.items():
+            if (name not in bound.arguments and
+                    param.default is not inspect._empty):
+                bound.arguments[name] = param.default
```

should this raise if there is a missing value + no default?